### PR TITLE
Use speedy XOR from pion/transport

### PR DESCRIPTION
**Pull request by jech**
Now that we have a speedy version of xorBytes in pion/transport,
we might as well use it.  Benchmarks indicate an improvement
of 10% to 20% in CTR encryption speed for large blocks, depending
on the architecture.

Related to https://github.com/pion/transport/commit/f0728d9ef5c3348561b849721055b109ac5c91d2. CC @adriancable

### CODE REVIEW

File: crypto.go [hunk 1]
```diff
@@ -14,12 +16,7 @@ func xorBytes(dst, src1, src2 []byte) int {
 	if len(dst) < n {
 		n = len(dst)
 	}
-
```
[the remark this answers is not in the document]
Comment (adriancable) [reply crypto.go:17]: Also @jech the length check is not needed, since this is already done in xor.XorBytes.

**Review by adriancable — CHANGES_REQUESTED** [review]

**Comment by jech** [comment]
The semantics of xorBytes and your version is not quite the same — your version panics if the destination is too short, while the version here truncates the output.  The simplest solution was to write a version of xorBytes that uses your code while exactly duplicating the semantics of the old version.

I can have a look to see if we can do without going through an intermediate function, but it will require carefully examining all call sites.

**Comment by adriancable** [comment]
@jech - I'm probably OK with the double-call. Can you post some benchmarks here? It's always good to have benchmarks in the PR.

**Comment by jech** [comment]
@adriancable you were right, the indirection was not needed.  Superseded by #211.

**Pull request closed without merging**